participant/app/routes/$localAgency/recertify/name.tsx

```diff
@@ -4,13 +4,68 @@ import { Trans } from "react-i18next";
 import { NameInput } from "~/components/NameInput";
 import type { NameInputProps } from "~/components/NameInput";
 import { RequiredQuestionStatement } from "~/components/RequiredQuestionStatement";
-import { ValidatedForm } from "remix-validated-form";
+import {
+  ValidatedForm,
+  setFormDefaults,
+  validationError,
+} from "remix-validated-form";
 import { representativeNameSchema } from "app/utils/validation";
 import { withZod } from "@remix-validated-form/with-zod";
+import { json, redirect } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import type { Params } from "@remix-run/react";
+import type { LoaderFunction } from "@remix-run/node";
+import { cookieParser } from "app/cookies.server";
+import type { RepresentativeNameData } from "app/types";
+import { routeFromName } from "~/utils/routing";
+import {
+  upsertSubmissionForm,
+  findSubmissionFormData,
+} from "app/utils/db.server";
 
 const representativeNameValidator = withZod(representativeNameSchema);
 
-export default function Index() {
+export const loader: LoaderFunction = async ({
+  request,
+  params,
+}: {
+  request: Request;
+  params: Params<string>;
+}) => {
+  const { submissionID, headers } = await cookieParser(request, params);
+  const existingNameData = (await findSubmissionFormData(
+    submissionID,
+    "name"
+  )) as RepresentativeNameData;
+  return json(
+    {
+      submissionID: submissionID,
+      ...setFormDefaults("representativeNameForm", existingNameData),
+    },
+    { headers: headers }
+  );
+};
+
+type loaderData = Awaited<ReturnType<typeof loader>>;
+
+export const action = async ({ request }: { request: Request }) => {
+  const formData = await request.formData();
+  const validationResult = await representativeNameValidator.validate(formData);
+  if (validationResult.error) {
+    console.log(`Validation error: ${validationResult.error}`);
+    return validationError(validationResult.error, validationResult.data);
+  }
+  const parsedForm = representativeNameSchema.parse(formData);
+  const { submissionID } = await cookieParser(request);
+  console.log(`Got submission ${JSON.stringify(parsedForm)}`);
+  await upsertSubmissionForm(submissionID, "name", parsedForm);
+  const routeTarget = routeFromName(request);
+  console.log(`Completed name form; routing to ${routeTarget}`);
+  return redirect(routeTarget);
+};
+
+export default function Namex() {
```

Comment:
Is this `Namex()` instead of `Name()` because `Name()` is a reserved function name?

Reply:
Potentially it's a typo... I just noticed it said `Index()` which didn't seem right. Are you suggesting `Name()` wouldn't work?

Reply:
No, I was not suggesting `Name()` wouldn't work. I was just curious why it was `Namex()` and speculating